site/src/pages/LoginPage/LoginPage.tsx

```diff
@@ -41,42 +49,19 @@ export const LoginPage: FC = () => {
 		});
 	}, [isSignedIn, buildInfoQuery.data, user?.id]);
 
-	if (isSignedIn) {
-		if (buildInfoQuery.data) {
-			// This uses `navigator.sendBeacon`, so window.href
-			// will not stop the request from being sent!
-			sendDeploymentEvent(buildInfoQuery.data, {
-				type: "deployment_login",
-				user_id: user?.id,
-			});
-		}
-
-		// If the redirect is going to a workspace application, and we
-		// are missing authentication, then we need to change the href location
-		// to trigger a HTTP request. This allows the BE to generate the auth
-		// cookie required.  Similarly for the OAuth2 exchange as the authorization
-		// page is served by the backend.
-		// If no redirect is present, then ignore this branched logic.
-		if (redirectTo !== "" && redirectTo !== "/") {
-			try {
-				// This catches any absolute redirects. Relative redirects
-				// will fail the try/catch. Subdomain apps are absolute redirects.
-				const redirectURL = new URL(redirectTo);
-				if (redirectURL.host !== window.location.host) {
-					window.location.href = redirectTo;
-					return null;
-				}
-			} catch {
-				// Do nothing
-			}
-			// Path based apps and OAuth2.
-			if (redirectTo.includes("/apps/") || redirectTo.includes("/oauth2/")) {
-				window.location.href = redirectTo;
-				return null;
-			}
-		}
-
-		return <Navigate to={redirectTo} replace />;
+	// The reason we need `window.location.href` for api redirects is that we
+	// need the page to reload and make a request to the backend. If we use
+	// `<Navigate>` react would handle the redirect itself and never request the
+	// page from the backend.
+	if (isSignedIn && isApiRouteRedirect) {
+		const sanitizedUrl = new URL(redirectTo, window.location.origin);
```

Comment:
This does not work for subdomain apps though right? If they are on another domain?

Reply:
They're not on another domain. The redirect from a workspace proxy subdomain app is to `/api/v2/applications/auth-redirect`. The redirect sanitization happens there.

Reply:
I'll trust you :+1:

Reply:
I could be wrong but that's what I always noticed in my testing.